lib/configuration.dart

```diff
@@ -0,0 +1,136 @@
+  late String instanceName;
+  bool optOut;
+  LogLevel logLevel;
+  int? minIdLength;
+  String? partnerId;
+  int flushMaxRetries;
+  bool useBatch;
+  ServerZone serverZone;
+  String? serverUrl;
+  int minTimeBetweenSessionsMillis;
+  DefaultTrackingOptions defaultTracking;
+  TrackingOptions trackingOptions;
+  MobileConfiguration mobileConfiguration;
+  AndroidConfiguration androidConfiguration;
+  WebConfiguration webConfiguration;
+  
+  Configuration({
+    required this.apiKey,
+    this.flushQueueSize = Constants.flushQueueSize,
+    this.flushIntervalMillis = Constants.flushIntervalMillis,
+    String instanceName = '',
+    this.optOut = false,
+    this.logLevel = LogLevel.warn,
+    this.minIdLength,
+    this.partnerId,
+    this.flushMaxRetries = Constants.flushMaxRetries,
+    this.useBatch = false,
+    this.serverZone = ServerZone.US,
+    this.serverUrl,
+    this.minTimeBetweenSessionsMillis = Constants.minTimeBetweenSessionsMillis,
+    this.defaultTracking = const DefaultTrackingOptions(),
+    TrackingOptions? trackingOptions,
+    this.mobileConfiguration = const MobileConfiguration(),
+    this.androidConfiguration = const AndroidConfiguration(),
+    this.webConfiguration = const WebConfiguration(),
+  }): trackingOptions = trackingOptions ?? TrackingOptions() {
+    this.instanceName = instanceName.isEmpty ? Constants.defaultInstanceName : instanceName;
+  }
+
+  Map<String, dynamic> toMap() {
+    return {
+      'apiKey': apiKey,
+      'flushQueueSize': flushQueueSize,
+      'flushIntervalMillis': flushIntervalMillis,
+      'instanceName': instanceName,
+      'optOut': optOut,
+      'logLevel': logLevel.toString(),
+      'minIdLength': minIdLength,
+      'partnerId': partnerId,
+      'flushMaxRetries': flushMaxRetries,
+      'useBatch': useBatch,
+      'serverZone': serverZone.toString(),
+      'serverUrl': serverUrl,
+      'minTimeBetweenSessionsMillis': minTimeBetweenSessionsMillis,
+      'defaultTracking': defaultTracking.toMap(),
+      'trackingOptions': trackingOptions.toMap(),
+      ...mobileConfiguration.toMap(),
+      ...androidConfiguration.toMap(),
+      ...webConfiguration.toMap(),
+    };
+  }
+}
+
+/// Configuration specific for iOS and Android devices.
+class MobileConfiguration {
+  final bool enableCoppaControl;
+  final bool flushEventsOnClose;
+  final int identifyBatchIntervalMillis;
+  final bool migrateLegacyData;
+
+  const MobileConfiguration({
+    this.enableCoppaControl = false,
+    this.flushEventsOnClose = true,
+    this.identifyBatchIntervalMillis = Constants.identifyBatchIntervalMillis,
+    this.migrateLegacyData = true,
+  });
+
+  Map<String, dynamic> toMap() {
+    return {
+      'enableCoppaControl': enableCoppaControl,
+      'flushEventsOnClose': flushEventsOnClose,
+      'identifyBatchIntervalMillis': identifyBatchIntervalMillis,
+      'migrateLegacyData': migrateLegacyData,
+    };
+  }
+}
+
+/// Configuration specific for Android devices.
```

Comment:
Curios do we have special for iOS side?

Reply:
No. iOS config is the union set of the general `Configuration` and `MobileConfiguration`